tests/components/tuya/test_init.py

```diff
@@ -16,29 +15,36 @@
 from tests.common import MockConfigEntry, async_load_json_object_fixture
 
 
-@pytest.mark.parametrize("mock_device_code", ["ydkt_jevroj5aguwdbs2e"])
-async def test_unsupported_device(
+async def test_device_registry(
     hass: HomeAssistant,
     mock_manager: ManagerCompat,
     mock_config_entry: MockConfigEntry,
-    mock_device: CustomerDevice,
+    mock_devices: CustomerDevice,
     device_registry: dr.DeviceRegistry,
     entity_registry: er.EntityRegistry,
     snapshot: SnapshotAssertion,
 ) -> None:
-    """Test unsupported device."""
+    """Validate device registry snapshots for all devices, including unsupported ones."""
 
-    await initialize_entry(hass, mock_manager, mock_config_entry, mock_device)
+    await initialize_entry(hass, mock_manager, mock_config_entry, mock_devices)
 
-    # Device is registered
-    assert (
-        dr.async_entries_for_config_entry(device_registry, mock_config_entry.entry_id)
-        == snapshot
-    )
-    # No entities registered
-    assert not er.async_entries_for_config_entry(
-        entity_registry, mock_config_entry.entry_id
+    device_registry_entries = dr.async_entries_for_config_entry(
+        device_registry, mock_config_entry.entry_id
     )
+    assert sorted(device_registry_entries, key=lambda x: x.model_id) == snapshot
```

Comment:
I think I might prefer having a snapshot per device, although it being sorted makes things easier